src/sql/workbench/api/node/extHostModelView.ts

```diff
@@ -27,59 +28,76 @@ class ModelBuilderImpl implements sqlops.ModelBuilder {
-		return this.withEventHandler(new CardWrapper(this._proxy, this._handle, id), id);
+		let builder: ComponentBuilderImpl<sqlops.CardComponent> = this.withEventHandler(new CardWrapper(this._proxy, this._handle, id), id);
+		this._components.set(id, builder.componentWrapper());
+		return builder;
 	}
 
 	inputBox(): sqlops.ComponentBuilder<sqlops.InputBoxComponent> {
 		let id = this.getNextComponentId();
-		return this.withEventHandler(new InputBoxWrapper(this._proxy, this._handle, id), id);
+		let builder: ComponentBuilderImpl<sqlops.InputBoxComponent> = this.withEventHandler(new InputBoxWrapper(this._proxy, this._handle, id), id);
+		this._components.set(id, builder.componentWrapper());
+		return builder;
 	}
 
 	checkBox(): sqlops.ComponentBuilder<sqlops.CheckBoxComponent> {
 		let id = this.getNextComponentId();
-		return this.withEventHandler(new CheckBoxWrapper(this._proxy, this._handle, id), id);
+		let builder: ComponentBuilderImpl<sqlops.CheckBoxComponent> = this.withEventHandler(new CheckBoxWrapper(this._proxy, this._handle, id), id);
+		this._components.set(id, builder.componentWrapper());
+		return builder;
 	}
 
 	button(): sqlops.ComponentBuilder<sqlops.ButtonComponent> {
 		let id = this.getNextComponentId();
-		return this.withEventHandler(new ButtonWrapper(this._proxy, this._handle, id), id);
+		let builder: ComponentBuilderImpl<sqlops.ButtonComponent> = this.withEventHandler(new ButtonWrapper(this._proxy, this._handle, id), id);
+		this._components.set(id, builder.componentWrapper());
+		return builder;
 	}
 
 	dropDown(): sqlops.ComponentBuilder<sqlops.DropDownComponent> {
 		let id = this.getNextComponentId();
-		return this.withEventHandler(new DropDownWrapper(this._proxy, this._handle, id), id);
+		let builder: ComponentBuilderImpl<sqlops.DropDownComponent> = this.withEventHandler(new DropDownWrapper(this._proxy, this._handle, id), id);
+		this._components.set(id, builder.componentWrapper());
+		return builder;
 	}
 
 	dashboardWidget(widgetId: string): sqlops.ComponentBuilder<sqlops.WidgetComponent> {
 		let id = this.getNextComponentId();
-		return this.withEventHandler<sqlops.WidgetComponent>(new ComponentWrapper(this._proxy, this._handle, ModelComponentTypes.DashboardWidget, id), id);
+		let builder = this.withEventHandler<sqlops.WidgetComponent>(new ComponentWrapper(this._proxy, this._handle, ModelComponentTypes.DashboardWidget, id), id);
+		this._components.set(id, builder.componentWrapper());
+		return builder;
 	}
 
 	dashboardWebview(webviewId: string): sqlops.ComponentBuilder<sqlops.WebviewComponent> {
 		let id = this.getNextComponentId();
-		return this.withEventHandler(new ComponentWrapper(this._proxy, this._handle, ModelComponentTypes.DashboardWebview, id), id);
+		let builder: ComponentBuilderImpl<sqlops.WebviewComponent> = this.withEventHandler(new ComponentWrapper(this._proxy, this._handle, ModelComponentTypes.DashboardWebview, id), id);
+		this._components.set(id, builder.componentWrapper());
```

Comment:
we don't need two maps to add component to, can you remove the other map and make it use this one?